.github/workflows/release.yml

```diff
@@ -141,3 +141,31 @@ jobs:
       run: |
         cd build
         gh release upload ${{ github.event.release.tag_name }} bundles/*
+
+  upload:
+    name: Upload
+    needs:
+    - source
+    - windows
+
+    runs-on: ubuntu-latest
+
+    # This job is empty, but ensures no upload job starts before all targets finished and are successful.
+    steps:
+    - name: Build completed
+      run: |
+        true
+
+  upload-cdn:
+    name: Upload (CDN)
+    needs:
+    - source
+    - upload
+
+    uses: ./.github/workflows/upload-cdn.yml
+    secrets: inherit
+
+    with:
+      version: ${{ needs.source.outputs.version }}
+      folder: ${{ needs.source.outputs.folder }}
+      trigger_type: ${{ needs.source.outputs.trigger_type }}
```

Comment:
From what I can tell, the `source` job in this workflow doesn't define either of those three. For reference:

https://github.com/OpenTTD/OpenTTD/blob/master/.github/workflows/release-source.yml#L145

It isn't completely trivial to create the right variables, as it needs to detect whether it is a nightly or release.